Bring docs/upload-your-snap.md in line with http://snapcraft.io/docs/…

dholbach opened the pull request:
…build-snaps/publish

LP: #1633080

sergiusens commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,157 @@
-# Upload to the store
+---
+title: "Publish your snap"
```
Maybe "Release your snap"

sergiusens commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,157 @@
+* by visiting: [the name registration page](https://myapps.developer.ubuntu.com/dev/click-apps/register-name/)
+* or by running `snapcraft register snap-name`
+
+
```
There seems to be extra line breaks above and below the Example title

sergiusens commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,157 @@
+form. If the name is already taken, the form will let you submit a name
+dispute.
+
+## 3. Upload your snap
```
Push your snap

dholbach commented:
@davcal - can you weigh in?

dholbach commented:
I changed the text to use *push* and *release* as terms and removed the extra newlines. I'll file a separate issue about documenting `snapcraft push ... --release`.

elopio commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,154 @@
 
-First of all, you need to give Snapcraft permission to communicate with the
-store on your behalf:
+Note that `logout` commands are available as well.
```
It's just one. So it should be: Note that the `logout` command is available as well.

caldav replied:
snapcraft logout and snap logout, no?

elopio replied:
ahh I see.

elopio commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,154 @@
+willingly accept that the snap is breaking out of confinement.
+
+### Pushing the snap
+Snap upload to the store can be done either from the Ubuntu Store web 
```
s/upload/push

dholbach replied:
"Snap push to the store" sounds a bit weird. Let me see what we can do here.

dholbach replied:
Fixed it. Thanks again.

elopio commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,154 @@
+
+### Revisions
+
+Each time you release a snap, the store will assigned a revision number to it,
```
s/assigned/assign

dholbach replied:
Thanks. Fixed.

caldav commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,154 @@
-essentially tags the resulting snap for life with the grade it should have.
+Although, snaps are not tied to a specific type of store and you can host
+them any way you want, see the [Store](/docs/core/store) page for more
+details.
```
Analytics on snapcraft.io show they don't get lost and this gives the context they need to understand store accounts and have an overivew of how the store works. Although, the link should be changed to a link to snapcraft.io (http://snapcraft.io/docs/core/store), since /docs/core/store won't work here.

dholbach replied:
Thanks. Fixed.

caldav commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,157 @@
-upload it directly to the store.
 
-## The grade option
+In order to share your snaps with the world, you need to publish them in a
```
Ok, but let's not forget that "publishing software" is the bread and butter of a lot of companies we are targeting with this content.

dholbach commented on docs/upload-your-snap.md:
```diff
@@ -1,69 +1,154 @@
+---
+title: "Release your snap"
+---
 
```
@caldav ^ Looks like this doesn't work (seehttps://github.com/dholbach/snapcraft/blob/4c63797af584a23a9027d399380cb1bbe04b8ef5/docs/upload-your-snap.md). Is this important for snapcraft.io? I copied it from there.

dholbach commented:
I filed https://bugs.launchpad.net/snapcraft/+bug/1633507 about documenting `snapcraft push --release`.
